2022-04-08 10:44:15,787 INFO  ==============================================================
2022-04-08 10:44:15,788 INFO  Id:    SoftwareManagement.sm_apt_install_fail
2022-04-08 10:44:15,789 INFO  Dir:   software_management_end_to_end/
2022-04-08 10:44:15,790 INFO  Title: Validate end to end behaviour for a failing installation
2022-04-08 10:44:15,791 DEBUG Execution order hint: 0.0
2022-04-08 10:44:15,791 INFO  ==============================================================
2022-04-08 10:44:15,797 INFO  Using sw id database: {'asciijump': '5475369', 'robotfindskitten': '5474869', 'squirrel3': '5475279', 'rolldice': '5152439', 'moon-buggy': '5439204', 'apple': '5495053', 'banana': '5494888', 'cherry': '5495382', 'watermelon': '5494510', 'registry': '8018911', 'hello-world': '8021526', 'docker/getting-started': '8021973', 'alpine': '7991792'}
2022-04-08 10:44:15,798 DEBUG EnvironmentC8y Setup
2022-04-08 10:44:15,799 INFO  EnvironmentC8y Setup
2022-04-08 10:44:15,801 DEBUG Process parameters for systemctl<serv_mapper1>
  command line : /usr/bin/systemctl status tedge-mapper-c8y
    arg #1     : status
    arg #2     : tedge-mapper-c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper1.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper1.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:44:15,808 DEBUG Waiting up to 600 secs for process systemctl<serv_mapper1> (pid 20670)
2022-04-08 10:44:16,366 INFO  Executed systemctl<serv_mapper1>, exit status 3
2022-04-08 10:44:16,476 INFO  Restarting mosquitto too frequently in the last 10 seconds. It was only up for 4 seconds
2022-04-08 10:44:16,478 INFO  Delaying execution by 7 seconds
2022-04-08 10:44:23,488 DEBUG Process parameters for sudo<tedge_connect_c8y>
  command line : /usr/bin/sudo /usr/bin/tedge connect c8y
    arg #1     : /usr/bin/tedge
    arg #2     : connect
    arg #3     : c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:44:23,497 DEBUG Waiting up to 600 secs for process sudo<tedge_connect_c8y> (pid 20675)
2022-04-08 10:44:37,292 INFO  Executed sudo<tedge_connect_c8y>, exit status 0, duration 13 secs
2022-04-08 10:44:37,297 DEBUG Process parameters for sudo<tedge_connect_c8y_test>
  command line : /usr/bin/sudo /usr/bin/tedge connect c8y --test
    arg #1     : /usr/bin/tedge
    arg #2     : connect
    arg #3     : c8y
    arg #4     : --test
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y_test.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y_test.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:44:37,317 DEBUG Waiting up to 600 secs for process sudo<tedge_connect_c8y_test> (pid 20781)
2022-04-08 10:44:37,941 INFO  Executed sudo<tedge_connect_c8y_test>, exit status 0
2022-04-08 10:44:37,944 DEBUG Process parameters for systemctl<serv_mosq2>
  command line : /usr/bin/systemctl status mosquitto
    arg #1     : status
    arg #2     : mosquitto
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mosq2.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mosq2.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:44:37,952 DEBUG Waiting up to 600 secs for process systemctl<serv_mosq2> (pid 20799)
2022-04-08 10:44:38,409 INFO  Executed systemctl<serv_mosq2>, exit status 0
2022-04-08 10:44:38,412 DEBUG Process parameters for systemctl<serv_mapper3>
  command line : /usr/bin/systemctl status tedge-mapper-c8y
    arg #1     : status
    arg #2     : tedge-mapper-c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper3.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper3.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:44:38,420 DEBUG Waiting up to 600 secs for process systemctl<serv_mapper3> (pid 20802)
2022-04-08 10:44:38,878 INFO  Executed systemctl<serv_mapper3>, exit status 0
2022-04-08 10:44:40,905 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:41,090 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations?deviceId=17064152&pageSize=1&revert=true&dateFrom=1970-01-01T00%3A00%3A00.000Z HTTP/1.1" 200 395
2022-04-08 10:44:41,104 DEBUG Final URL of the request: https://thin-edge-io.eu-latest.cumulocity.com/devicecontrol/operations?deviceId=17064152&pageSize=1&revert=true&dateFrom=1970-01-01T00%3A00%3A00.000Z
2022-04-08 10:44:41,108 ERROR No operations found, assuming it passed
2022-04-08 10:44:41,109 DEBUG --- test execute
2022-04-08 10:44:41,126 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:41,322 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "POST /devicecontrol/operations HTTP/1.1" 201 488
2022-04-08 10:44:41,338 INFO  Response status: 201
2022-04-08 10:44:41,340 INFO  Response to action: {
    "creationTime": "2022-04-08T10:44:41.296Z",
    "deviceId": "17064152",
    "deviceName": "offsite_mythicd",
    "self": "https://t493319102.eu-latest.cumulocity.com/devicecontrol/operations/17065147",
    "id": "17065147",
    "status": "PENDING",
    "description": "Apply software changes, triggered from PySys: [{'name': 'does_not_exist', 'id': '5446165', 'version': '::apt', 'url': '', 'action': 'install'}]",
    "c8y_SoftwareUpdate": [
        {
            "name": "does_not_exist",
            "action": "install",
            "id": "5446165",
            "version": "::apt",
            "url": ""
        }
    ]
}
2022-04-08 10:44:41,341 INFO  Started operation: {'creationTime': '2022-04-08T10:44:41.296Z', 'deviceId': '17064152', 'deviceName': 'offsite_mythicd', 'self': 'https://t493319102.eu-latest.cumulocity.com/devicecontrol/operations/17065147', 'id': '17065147', 'status': 'PENDING', 'description': "Apply software changes, triggered from PySys: [{'name': 'does_not_exist', 'id': '5446165', 'version': '::apt', 'url': '', 'action': 'install'}]", 'c8y_SoftwareUpdate': [{'name': 'does_not_exist', 'action': 'install', 'id': '5446165', 'version': '::apt', 'url': ''}]}
2022-04-08 10:44:43,360 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:43,557 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:43,573 INFO  State of operation 17065147 : PENDING
2022-04-08 10:44:45,592 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:45,773 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:45,789 INFO  State of operation 17065147 : PENDING
2022-04-08 10:44:48,883 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:49,366 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:49,385 INFO  State of operation 17065147 : PENDING
2022-04-08 10:44:51,430 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:52,137 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:52,158 INFO  State of operation 17065147 : PENDING
2022-04-08 10:44:54,178 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:54,361 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:54,378 INFO  State of operation 17065147 : PENDING
2022-04-08 10:44:56,416 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:56,670 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:56,692 INFO  State of operation 17065147 : PENDING
2022-04-08 10:44:58,715 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:44:58,911 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 640
2022-04-08 10:44:58,947 INFO  State of operation 17065147 : PENDING
2022-04-08 10:45:00,969 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:01,185 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:01,208 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:03,236 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:03,423 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:03,440 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:05,459 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:05,641 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:05,657 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:07,676 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:07,858 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:07,874 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:09,893 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:10,072 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:10,088 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:12,110 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:12,315 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:12,332 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:14,354 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:15,025 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:15,040 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:17,060 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:17,825 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:17,842 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:19,864 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:20,176 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:20,200 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:22,228 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:22,409 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:22,425 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:24,444 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:24,629 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:24,646 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:26,665 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:26,841 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:26,858 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:28,880 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:29,069 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:29,086 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:31,108 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:31,326 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 642
2022-04-08 10:45:31,351 INFO  State of operation 17065147 : EXECUTING
2022-04-08 10:45:33,370 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:33,755 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /devicecontrol/operations/17065147 HTTP/1.1" 200 758
2022-04-08 10:45:33,770 INFO  State of operation 17065147 : FAILED
2022-04-08 10:45:33,771 DEBUG --- test validate
2022-04-08 10:45:33,788 DEBUG Starting new HTTPS connection (1): thin-edge-io.eu-latest.cumulocity.com:443
2022-04-08 10:45:34,396 DEBUG https://thin-edge-io.eu-latest.cumulocity.com:443 "GET /inventory/managedObjects/17064152 HTTP/1.1" 200 1336
2022-04-08 10:45:34,411 WARN  TypeError occurred while running test - 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/home/pi/env-pysys/lib/python3.7/site-packages/pysys/baserunner.py", line 1296, in __call__
    self.testObj.validate()
  File "/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/run.py", line 31, in validate
    "False == value", value=self.check_is_installed("does_not_exist")
  File "/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/environment_sm_management.py", line 398, in check_is_installed
    for package in package_list:
TypeError: 'NoneType' object is not iterable
2022-04-08 10:45:34,439 WARN  TypeError: 'NoneType' object is not iterable ... blocked 
2022-04-08 10:45:34,440 DEBUG --- test cleanup
2022-04-08 10:45:34,441 INFO  
2022-04-08 10:45:34,442 INFO  cleanup:
2022-04-08 10:45:34,443 DEBUG Running registered cleanup function: <bound method SoftwareManagement.mysmcleanup of <PySysTest object at 0x75db7470>>
2022-04-08 10:45:34,443 DEBUG Running registered cleanup function: <bound method EnvironmentC8y.myenvcleanup of <PySysTest object at 0x75db7470>>
2022-04-08 10:45:34,444 DEBUG EnvironmentC8y Cleanup
2022-04-08 10:45:34,569 DEBUG Process parameters for sudo<tedge_disconnect_c8y>
  command line : /usr/bin/sudo /usr/bin/tedge disconnect c8y
    arg #1     : /usr/bin/tedge
    arg #2     : disconnect
    arg #3     : c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_disconnect_c8y.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_disconnect_c8y.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:45:34,577 DEBUG Waiting up to 600 secs for process sudo<tedge_disconnect_c8y> (pid 20985)
2022-04-08 10:45:38,512 INFO  Executed sudo<tedge_disconnect_c8y>, exit status 0
2022-04-08 10:45:38,515 DEBUG Process parameters for systemctl<serv_mapper5>
  command line : /usr/bin/systemctl status tedge-mapper-c8y
    arg #1     : status
    arg #2     : tedge-mapper-c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper5.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper5.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 10:45:38,523 DEBUG Waiting up to 600 secs for process systemctl<serv_mapper5> (pid 21026)
2022-04-08 10:45:39,080 INFO  Executed systemctl<serv_mapper5>, exit status 3
2022-04-08 10:45:39,081 DEBUG ProcessUser cleanup function done.
2022-04-08 10:45:39,085 INFO  
2022-04-08 10:45:39,086 INFO  Test duration: 83.31 secs
2022-04-08 10:45:39,087 INFO  Test final outcome:  BLOCKED
2022-04-08 10:45:39,088 INFO  Test outcome reason: TypeError: 'NoneType' object is not iterable
2022-04-08 10:45:39,089 INFO  
